src/StreamSelectLoop.php

```diff
@@ -287,8 +287,28 @@ private function streamSelect(array &$read, array &$write, $timeout)
                 }
             }
 
-            // suppress warnings that occur, when stream_select is interrupted by a signal
-            $ret = @\stream_select($read, $write, $except, $timeout === null ? null : 0, $timeout);
+            /** @var ?callable $previous */
+            $previous = \set_error_handler(function ($errno, $errstr) use (&$previous) {
+                // suppress warnings that occur when `stream_select()` is interrupted by a signal
+                $eintr = \defined('SOCKET_EINTR') ? \SOCKET_EINTR : 4;
+                if ($errno === \E_WARNING && \strpos($errstr, '[' . $eintr .']: ') !== false) {
+                    return;
+                }
+
+                // forward any other error to registered error handler or print warning
+                return ($previous !== null) ? \call_user_func_array($previous, \func_get_args()) : false;
+            });
+
+            try {
+                $ret = \stream_select($read, $write, $except, $timeout === null ? null : 0, $timeout);
+                \restore_error_handler();
+            } catch (\Throwable $e) { // @codeCoverageIgnoreStart
+                \restore_error_handler();
+                throw $e;
+            } catch (\Exception $e) {
```

Comment:
Exception extends Throwable so probably no needs for second catch block

Reply:
PHP 5.3 support, same as above [above](https://github.com/reactphp/event-loop/pull/245#discussion_r812352705).